File: leetcode/leetcode-patterns/remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
  def deleteDuplicates(self, head: ListNode) -> ListNode:
    curr = head
    while curr is not None:
      while curr.next is not None and curr.next.val == curr.val:
        curr.next = curr.next.next
      curr = curr.next
    return head

File: leetcode/leetcode-patterns/remove-duplicates-from-sorted-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
 var deleteDuplicates = function(head) {
  let dummy = new ListNode();
  let curr = head;
  let new_head = dummy;
  
  while (curr !== null) {
    if (curr.next == null || curr.val !== curr.next.val) {
      dummy.next = new ListNode(curr.val);
      dummy = dummy.next;
    }
    curr = curr.next;
  }
  
  return new_head.next;
};